src/FakeItEasy/Core/FakeManager.cs

```diff
@@ -182,6 +182,17 @@ internal virtual void ClearUserRules()
         }
 
         /// <summary>
+        /// Removes any recorded calls.
+        /// </summary>
+        internal virtual void ClearRecordedCalls()
+        {
+            ICompletedFakeObjectCall ignored;
+            while (this.recordedCalls.TryDequeue(out ignored))
+            {
+            }

```

Comment:
Agreed. Replacing the queue with a new one is atomic, whereas dequeueing all items is not.